crates/napi/src/bindgen_runtime/js_values/either.rs

```diff
@@ -0,0 +1,318 @@
+use super::{FromNapiValue, ToNapiValue, TypeName};
+use crate::{type_of, Status, ValueType};
+
+#[derive(Debug, Clone, Copy)]
+pub enum Either<
```

Comment:
macro?

Reply:
I've tried macro, it could not reduce codes from the `from_napi_value` parts, so I gave up.